lib/v1_alpha_fetcher.rb

```diff
@@ -0,0 +1,45 @@
+class V1AlphaFetcher
+  def self.fetch(url)
+    url += "/apis"
+
+    uri = URI(url)
+
+    req = Net::HTTP::Get.new(uri)
+    req["correlation-id"] = SecureRandom.uuid
+    req["accept"] = "application/vnd.uk.gov.api.v1alpha+json"
+
+    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
+      http.request(req)
+    end
+    raise "Could not parse APIs from #{url}, no version supported" if response.is_a? Net::HTTPNotAcceptable
+
+    raise "A client error caused retrieval of APIs from #{url} to fail" if response.is_a? Net::HTTPNotFound
+
+    response = JSON.parse response.body
+
+    apis = response["apis"].map do |api|
+      data = api["data"]
+      Api.new name: data["name"],
+              description: data["description"],
+              url: data["url"],
+              documentation: data["documentation-url"],
+              provider: data["organisation"].gsub(/\s+/, "-").downcase
```

Comment:
is there any reason this doesn't use `create_organisation`? This would leave to the `provider` having a different organisation name?

Reply:
I misread this - `id = name.gsub(/\s+/, "-").downcase` below matches this